### PR TITLE
Display new attachments markdown syntax

### PR DESCRIPTION
**Pull request by dnkrj**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

# What
- Render attachments in a list instead of a table
- Add Markdown snippets to the Attachments listing page
- Remove inline attachment markdown from edition form
- Use the new filename based attachments markdown
- Update the govspeak guidance to reflect the new syntax

# Why
- New syntax is easier to use
- Makes attachments and images tabs consistent
- Shortens edition form

https://trello.com/c/FZ5z3Me7/1270-add-markdown-snippets-to-the-attachments-listing-page
https://trello.com/c/r2D05jNN/1357-comms-guidance-updates-for-new-govspeak-attachment-syntax

# Screenshots
### After
whitehall-admin dev gov uk_government_admin_editions_1374132_attachments(iPad Pro) (3)

### Before
whitehall-admin dev gov uk_government_admin_editions_1374132_attachments(iPad Pro) (1)

### After
whitehall-admin dev gov uk_government_admin_news_1374132_edit(iPad Pro) (2)

### Before
whitehall-admin dev gov uk_government_admin_news_1374132_edit(iPad Pro) (1)

## What's new page
whitehall-admin dev gov uk_government_admin_whats-new(iPad Pro) (1)

### CODE REVIEW

File: app/views/admin/attachments/_attachments.html.erb [hunk 1]
```diff
@@ -2,37 +2,41 @@
...
+        </p>
+        <% if attachable.allows_inline_attachments? %>
+          <%= render "govuk_publishing_components/components/copy_to_clipboard", {
+            label: tag.strong("Inline attachment markdown code:"),
+            copyable_content: "[AttachmentLink: #{attachment_component_params(attachment)[:id]}]",
+            button_text: "Copy Markdown"
+          } %>
+          <div class="app-view-edition-resource__markdown--padded">
+            <%= render "govuk_publishing_components/components/copy_to_clipboard", {
+              label: tag.strong("Publication box markdown code:"),
```
Comment (ollietreend) [inline app/views/admin/attachments/_attachments.html.erb:26]: I'm not sure this is the right labelling to use. I think we should at least try to be consistent with the Govspeak help sidebar, which calls it an "attachment as a callout box" and a "link to an attachment".

Or, how about we standardise on this terminology?

> **Embed this attachment:**
> ```
> [Attachment: example.pdf]
> ```
>
> **Link to this attachment:**
> ```
> [AttachmentLink: example.pdf]
> ```

File: app/views/admin/editions/_govspeak_attachments_help.html.erb [hunk 2]
```diff
@@ -5,10 +5,6 @@
...
 <p class="govuk-body">Add an attachment as a callout box:</p>
-<pre>!@<em>n</em></pre>
+<pre class="govspeak-help__pre">[Attachment: <em>filename.pdf</em>]</pre>
 <p class="govuk-body">Or, if you want to use a link to an attachment in a sentence or list:</p>
-<pre>[InlineAttachment:<em>n</em>]</pre>
-<p class="govuk-body">eg for the first file:</p>
-<pre>!@1</pre>
-<p class="govuk-body">or:</p>
-<pre>[InlineAttachment:1]</pre>
+<pre class="govspeak-help__pre">[AttachmentLink: <em>filename.pdf</em>]</pre>
```
Comment (ollietreend) [inline app/views/admin/editions/_govspeak_attachments_help.html.erb:10]: Carrying across that terminology, we update this help text so be:

> Embed an attachment:
> ```
> [Attachment: example.pdf]
> ```
>
> Link to an attachment:
> ```
> [AttachmentLink: example.pdf]
> ```

**Review by ollietreend — COMMENTED** [review]
Nice work! 😄 and thanks for tidying up the feature test helpers too.

Just a few opinions from me – no technical blockers though. Let me know your thoughts.

File: app/views/admin/attachments/_attachments.html.erb [hunk 3]
```diff
@@ -2,37 +2,41 @@
...
+<ul class="govuk-list">
+  <% attachable.attachments.includes(:attachment_data).map.with_index(1) do |attachment, index| %>
+    <li class="govuk-grid-row">
+      <div class="govuk-grid-column-full">
+        <p class="govuk-body">
+          <strong>Title:</strong> <%= attachment.title %> <% unless attachment.attachment_data.blank? || attachment.attachment_data.uploaded_to_asset_manager? %><span class="govuk-tag govuk-tag--green">Uploading</span><% end %>
+        </p>
+        <p class="govuk-body">
+          <strong>Type: </strong><%= attachment.is_a?(HtmlAttachment) ? attachment.readable_type : attachment.readable_type.capitalize %>
+        </p>
```
[the remark this answers is not in the document]
Comment (dnkrj) [reply app/views/admin/attachments/_attachments.html.erb:14]: Yeah, it's a good point
I think that we'll be most likely be moving to the [summary card component](https://design-system.service.gov.uk/components/summary-list/#summary-cards) once the transition team have more fully established its use in Whitehall.

File: app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -23,7 +23,7 @@ $govuk-page-width: 1140px;
 @import "./admin/views/document-history-tab";
 @import "./admin/views/edit-edition";
 @import "./admin/views/edit-person";
-@import "./admin/views/edition-images";
+@import "./admin/views/edition-resource";
```
[the remark this answers is not in the document]
Comment (dnkrj) [reply app/assets/stylesheets/application.scss:26]: I chose "resource" based on rails routing in the case that we could add other tabs for things that are not assets - tags for example and would want some of the same patterns.

But agree for just images and attachments "assets" makes more sense.

What do you think? More than happy to change it to assets for time being, how likely do you think that we are to add other tabs?

[the remark this answers is not in the document]
Comment (dnkrj) [reply app/assets/stylesheets/application.scss:26]: So the options are `edition-resource`, `edition-asset` or `edition-tab`

[the remark this answers is not in the document]
Comment (ollietreend) [reply app/assets/stylesheets/application.scss:26]: Yeah, that seems reasonable.

I guess my initial reaction is that I wouldn't expect any other tabs to need to make use of CSS rules for 'preview' (`.app-view-edition-resource__preview`) or Markdown snippets (`.app-view-edition-resource__markdown--padded`). So this does feel very focussed on 'the user uploaded a thing, and now needs to embed it' – which I don't expect we'll be adding other 'types' of embeddable thing beyond images and file attachments.

But at the same time, my "asset" argument technically breaks down when it comes to HTML attachments and external attachments – neither of which hold a user uploaded file. 🤷🏻

So, I don't have a strong opinion I guess. Feel free to stick with 'resource'!

**Review by ollietreend — APPROVED** [review]
LGTM! ✨🚀

**Pull request merged**